### PR TITLE
feat: Add custom JSON schema errors

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery-issues/issues/1526 (internal issue).
This PR uses the custom errors syntax from https://ajv.js.org/packages/ajv-errors.html#messages-for-properties-and-items to improve the errors we provide for the S3 destination plugin (and aligns file and GCS plugin with it).
While this won't help with the Go validation until https://github.com/santhosh-tekuri/jsonschema/issues/44 is implemented it does help with our Cloud Syncs, examples below.

The `errorMessage->properties->prop-name` is needed so the UI knows to show the error under the correct field and not as a top level error

Before:
image

After:
image

Before:
image

After:
image

(The latter still need some frontend work not to show the top level error)

### CODE REVIEW

File: plugins/destination/file/client/spec/schema.go [hunk 1]
```diff
@@ -25,6 +25,22 @@ func (s Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
 			return properties
 		}(),
 	}
+
+	pathNotWithUUID := &jsonschema.Schema{
```
Comment (erezrokah) [inline plugins/destination/file/client/spec/schema.go:29]: Needed so we can assign the error via `errorMessage->properties->path`

File: plugins/destination/file/client/spec/spec.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ type Spec struct {
 	// - `{{MINUTE}}` will be replaced with the current minute in `mm` format
 	//
 	//  **Note** that timestamps are in `UTC` and will be the current time at the time the file is written, not when the sync started.
-	Path string `json:"path,omitempty" jsonschema:"required,minLength=1"`
+	Path string `json:"path,omitempty" jsonschema:"required,minLength=1,example=path/to/files/{{TABLE}}/{{UUID}}.parquet" jsonschema_extras:"errorMessage=value should not start with /"`
```
Comment (erezrokah) [inline plugins/destination/file/client/spec/spec.go:41]: Added missing example though this plugin is not enabled for Cloud Syncs

File: plugins/destination/file/docs/_configuration.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ spec:
   write_mode: "append"
   spec:
     path: "path/to/files/{{TABLE}}/{{UUID}}.{{FORMAT}}"
-    format: "csv" # options: parquet, json, csv
+    format: "parquet" # options: parquet, json, csv
```
Comment (erezrokah) [inline plugins/destination/file/docs/_configuration.md:13]: Aligns with S3

File: plugins/destination/gcs/client/spec/schema.go [hunk 4]
```diff
@@ -39,38 +39,46 @@ func (s Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
...
-					Title: "`path` contains `//`",
-					Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
-						properties := orderedmap.New[string, *jsonschema.Schema]()
-						properties.Set("path", &jsonschema.Schema{
-							Type:    "string",
-							Pattern: `^.*//.*$`,
-						})
-						return properties
-					}(),
+		Extras: map[string]interface{}{
```
Comment (erezrokah) [inline plugins/destination/gcs/client/spec/schema.go:57]: Simplify the schema and also allow using `errorMessage->properties->path` mapping

File: plugins/destination/gcs/client/spec/schema.go [hunk 5]
```diff
@@ -133,14 +162,21 @@ func (s Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
 						properties.Set("path", &jsonschema.Schema{
 							Not: &jsonschema.Schema{
 								Type:    "string",
-								Pattern: `^.*{{.*}}.*$`,
+								Pattern: `^.*\{\{.*\}\}.*$`,
```
Comment (erezrokah) [inline plugins/destination/gcs/client/spec/schema.go:165]: This regex works on Go apperatnly but fails in JavaScript, so we need to escape the brackets

File: plugins/destination/gcs/client/spec/spec.go [hunk 6]
```diff
@@ -29,8 +29,20 @@ type Spec struct {
 	// Bucket where to sync the files.
 	Bucket string `json:"bucket,omitempty" jsonschema:"required,minLength=1"`
 
-	// Path to where the files will be uploaded in the above bucket.
-	Path string `json:"path,omitempty" jsonschema:"required,minLength=1"`
+	// Path to where the files will be uploaded in the above bucket, for example `path/to/files/{{TABLE}}/{{UUID}}.parquet`
```
Comment (erezrokah) [inline plugins/destination/gcs/client/spec/spec.go:32]: Align with S3

File: plugins/destination/gcs/docs/_configuration.md [hunk 7]
```diff
@@ -10,8 +10,8 @@ spec:
   write_mode: "append"
   spec:
     bucket: "bucket_name"
-    path: "path/to/files"
-    format: "csv" # options: parquet, json, csv
+    path: "path/to/files/{{TABLE}}/{{UUID}}.{{FORMAT}}"
```
Comment (erezrokah) [inline plugins/destination/gcs/docs/_configuration.md:13]: Align with S3

**Comment by erezrokah** [comment]
cc @pilvikala for feedback on the result

**Comment by pilvikala** [comment]
Huge improvement, thank you!

**Review by disq — APPROVED** [review]

**Pull request merged**